pwiz_tools/Skyline/SkylineNightly/Nightly.cs

```diff
@@ -958,40 +958,41 @@ private string PostToLink(string link, string postData, string filePath)
                 wr.KeepAlive = true;
                 wr.Credentials = CredentialCache.DefaultCredentials;
 
-                SetCSRFToken(wr, LogFileName);
-
-                var rs = wr.GetRequestStream();
-
-                rs.Write(boundarybytes, 0, boundarybytes.Length);
-                const string headerTemplate = "Content-Disposition: form-data; name=\"{0}\"; filename=\"{1}\"\r\nContent-Type: {2}\r\n\r\n";
-                string header = string.Format(headerTemplate, "xml_file", filePath != null ? Path.GetFileName(filePath) : "xml_file", "text/xml");
-                byte[] headerbytes = Encoding.UTF8.GetBytes(header);
-                rs.Write(headerbytes, 0, headerbytes.Length);
-                var bytes = Encoding.UTF8.GetBytes(postData);
-                rs.Write(bytes, 0, bytes.Length);
-
-                byte[] trailer = Encoding.ASCII.GetBytes("\r\n--" + boundary + "--\r\n");
-                rs.Write(trailer, 0, trailer.Length);
-                rs.Close();
-
-                WebResponse wresp = null;
-                try
+                if (SetCSRFToken(wr, LogFileName))
                 {
-                    wresp = wr.GetResponse();
-                    var stream2 = wresp.GetResponseStream();
-                    if (stream2 != null)
+                    var rs = wr.GetRequestStream();
+
+                    rs.Write(boundarybytes, 0, boundarybytes.Length);
+                    const string headerTemplate = "Content-Disposition: form-data; name=\"{0}\"; filename=\"{1}\"\r\nContent-Type: {2}\r\n\r\n";
+                    string header = string.Format(headerTemplate, "xml_file", filePath != null ? Path.GetFileName(filePath) : "xml_file", "text/xml");
+                    byte[] headerbytes = Encoding.UTF8.GetBytes(header);
+                    rs.Write(headerbytes, 0, headerbytes.Length);
+                    var bytes = Encoding.UTF8.GetBytes(postData);
+                    rs.Write(bytes, 0, bytes.Length);
+
+                    byte[] trailer = Encoding.ASCII.GetBytes("\r\n--" + boundary + "--\r\n");
+                    rs.Write(trailer, 0, trailer.Length);
+                    rs.Close();
+
+                    WebResponse wresp = null;
+                    try
                     {
-                        var reader2 = new StreamReader(stream2);
-                        var result = reader2.ReadToEnd();
-                        return result;
+                        wresp = wr.GetResponse();
+                        var stream2 = wresp.GetResponseStream();
```

Comment:
I think in this context that gets complicated, since it's referenced in the catch block  - I expect it's OK to just let the GC deal with eventually. At any rate, it's been this way forever...